src/components/project_admin/ag_suites/ag_suites.vue

```diff
@@ -0,0 +1,608 @@
+    this.reset_active_indices();
+    this.test_suites.splice(index, 1);
+    if (suite_was_active) {
+      if (this.test_suites.length === 0) {
+        this.update_active_suite(null);
+      }
+      else if (index === this.test_suites.length) {
+        this.update_active_suite(this.test_suites[index - 1]);
+      }
+      else {
+        this.update_active_suite(this.test_suites[index]);
+      }
+    }
+  }
+
+  update_ag_test_suites_order_changed(project_pk: number, ag_test_suite_order: ID[]): void {}
+
+  // CaseObserver --------------------------------------------------------------------------------
+  update_ag_test_case_changed(ag_test_case: AGTestCase): void {
+    let parent_suite = this.test_suites[this.test_suites.findIndex(
+      (ag_suite: AGTestSuite) => ag_suite.pk === ag_test_case.ag_test_suite
+    )];
+    let case_index = parent_suite!.ag_test_cases.findIndex(
+      (test_case: AGTestCase) => test_case.pk === ag_test_case.pk
+    );
+    Vue.set(
+      parent_suite.ag_test_cases,
+      case_index,
+      ag_test_case
+    );
+  }
+
+  update_ag_test_case_created(ag_test_case: AGTestCase): void {
+    let parent_suite = this.test_suites[this.test_suites.findIndex(
+      (ag_suite: AGTestSuite) => ag_suite.pk === ag_test_case.ag_test_suite
+    )];
+    parent_suite.ag_test_cases.push(ag_test_case);
+    this.update_active_case(ag_test_case, parent_suite);
+  }
+
+  update_ag_test_case_deleted(ag_test_case: AGTestCase): void {
+    let parent_suite = this.test_suites[this.test_suites.findIndex(
+      (ag_suite: AGTestSuite) => ag_suite.pk === ag_test_case.ag_test_suite
+    )];
+    let case_index = parent_suite!.ag_test_cases.findIndex(
+      (test_case: AGTestCase) => test_case.pk === ag_test_case.pk
+    );
+    let case_was_active: boolean = this.active_case !== null
+                                   && ag_test_case.pk === this.active_case.pk;
+    this.reset_active_indices();
+    parent_suite!.ag_test_cases.splice(case_index, 1);
+    if (case_was_active) {
+      if (parent_suite.ag_test_cases.length === 0) {
+        this.update_active_suite(parent_suite!);
+      }
+      else if (case_index === parent_suite!.ag_test_cases.length) {
+       this.update_active_case(parent_suite!.ag_test_cases[case_index - 1],
+                               parent_suite!);
+      }
+      else {
+        this.update_active_case(parent_suite!.ag_test_cases[case_index],
+                                parent_suite!);
+      }
+    }
+  }
+
+  update_ag_test_cases_order_changed(ag_test_suite_pk: number, ag_test_case_order: number[]) { }
+
+  // CommandObserver------------------------------------------------------------------------------
+  update_ag_test_command_changed(ag_test_command: AGTestCommand): void {
+    let parent_suite: AGTestSuite = this.active_suite!;
+    let parent_case: AGTestCase = this.active_case!;
+
+    for (let suite_index = 0; suite_index < this.test_suites.length; ++suite_index) {
```

Comment:
Write a "find parent test case and suite" function to remove this duplicated loop